protos/dcs/mission/v0/mission.proto

```diff
@@ -310,6 +310,16 @@ message StreamEventsResponse {
     // what was typed
     string message = 2;
   }
+
+  // fired when the player changes across to a slot
+  message PlayerChangeSlotEvent {
```

Comment:
The handler method is TRYChangeSlot so should this event naming reflect that? is there a handler.OnPlayerSuccessChangeSlot or something that we can also emit an event for?

Reply:
Although we are intercepting the DCS's `try...` function, there is no equivalent "did" change slot; it's very similar motivation to how we intercept the `try...chat` message -- the caller expects an synchronous response (pure function) which would not be possible with the async trip the rust-server would need to make to the connected clients.

If no return value is made by the handler - as per the implementation - other hooks such as `slmod` would then execute. We just intercept, pass it up the gRPC clients.

If the gRPC determines it should not be allowed, they are welcome to execute the appropriate `ForcePlayerSlot`.

Reply:
Example use case .... https://github.com/ciribob/DCS-SimpleSlotBlock
My extended case will be around a few integration points (e.g. authentication, match organizing, long running missions)

Reply:
Right, but the event "PlayerChangeSlotEvent", at least to me, implies they _did_ change slot. `PlayerTryChangeSlotEvent` seems a more accurate event name.

Reply:
> `PlayerChangeSlotEvent`, at least to me, implies they did change slot. 

I agree with this as the premise "if it is in the past."

Based on the above proposed behavior, from the _gRPC clients' perspective (and timeline)_, it did occur by the time they read the streamed event.

Internally, the following _has_ to occur due to the synchronous behavior of `onPlayerTryChangeSlot`:
1. Read the message - make a decision. 
2. Since it cannot determine an outcome, it does not return a value (cannot make a decision).
3. It would enqueue the event that a player has selected a slot (by implication). This is the part which is in the past tense. 

When the gRPC Client does read the streamed events, the client has a good chance they have been accepted into the slot (assuming no other hook has evicted the person, or they person hasn't changed slots).

So therefore, when the client does read the event, it has occurred in the past. At this point, using a `ForcePlayerSlot` could force a player into a certain workflow (e.g. revoke/eject the player's slot selection).

Reply:
> I agree with this as the premise "if it is in the past."

> Based on the above proposed behavior, from the gRPC clients' perspective (and timeline), it did occur by the time they read the streamed event.

The client has definitely _tried_ to change slot by the time that the client receives the event but there is no guarantee that it succeeded _after_ the client has read the event since anything else in the handler chain could have denied the change.

This is why having the absolute "The player changed slot" is misleading I think. At the point that the DCS-gRPC client receives the event we have no idea if it succeeded or not. That is why I believe we should rename the event to highlight the fact that the player _tried_ to change slot at the point we receive the message but we have no idea if it was successful.

Reply:
Used different hook point.